sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -165,6 +165,7 @@ def connect(self):
             self._init_socket(
                 self.socket_settings, self.read_timeout, self.write_timeout,
             )
+            self.sock.settimeout(0.2)
```

Comment:
@annatisch - This seems to be a sweet spot for SSLTransport. Previously, the default timeout was 1, and this resulted in really slow receives when a partition was empty. I'm not sure whether this is the best solution to this, or a temporary one. When trying this out with Kashif, we realized that this "sweet spot" might change depending on the connection/latency. Is this reasonable for now? Do you know if there's a better way to resolve this?

Reply:
sounds good! just updated in the _transport.py file:
1) `TIMEOUT_INTERVAL = 0.2` - this will be used for SSLTransport now
2) `WS_TIMEOUT_INTERVAL = 1` - this has been added b/c this interval works for WebSocketTransport.

Reply:
The new constant does not appear to be used?
And it looks like it should be added to the `init_socket` method, with some kind of kwarg override.

Reply:
ah sorry, just passed `connect_timeout` into `init_socket` as a kwarg, and set that final `settimeout` in `init_socket` to connect_timeout.